Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditquerylog/AuditQueryLogImpl.java

```diff
@@ -50,35 +68,80 @@
      */
-    public QueryAuditEventsResponseType queryAuditEvents(QueryAuditEventsRequestType requestAuditEvents) {
-        return new QueryAuditEventsResponseType();
+    public QueryAuditEventsResponseType queryAuditEvents(QueryAuditEventsRequestType request) {
+        return resultUtil.getQueryAuditEventResponse(getAuditRepositoryDao().queryByAuditOptions(
+            getEventOutcome(request.getEventOutcomeIndicator()), getEventTypes(request.getEventTypeList()),
+            request.getUserId(), getRemoteHcids(request.getRemoteHcidList()), getRequestDate(request.getEventBeginDate()),
+            getRequestDate(request.getEventEndDate())));
+
     }
 
     /**
      *
-     * @param requestAuditEvents - Audit search params MessageId and RelatesTo will be provided by Request. These are
-     * optional fields in requestAuditEvents
+     * @param request - Audit search params MessageId and RelatesTo will be provided by Request. These are optional
+     * fields in requestAuditEvents
      * @return QueryAuditEventsResponseType - The Response will be having EventType or ServiceName, EventStatus- Success
      * or Failure, Event Timestamp, UserId(Human who initiated transaction), Direction (Outbound/Inbound), MessageID
      * -RequestMessageID, RelatesTo (Relates the DeferredRequests and DeferredResponses, Remote Organization Id and
      * Audit Id.
      *
-     * The Adapter logic will be implemented in FHAC-690
      */
-    public QueryAuditEventsResponseType queryAuditEventsByMessageId(
-        QueryAuditEventsRequestByRequestMessageId requestAuditEvents) {
-        return new QueryAuditEventsResponseType();
+    public QueryAuditEventsResponseType queryAuditEventsByMessageIdAndRelatesTo(
+        QueryAuditEventsRequestByRequestMessageId request) {
+        return resultUtil.getQueryAuditEventResponse(getAuditRepositoryDao().queryAuditRecords(
+            request.getRequestMessageId(), request.getRelatesTo()));
     }
 
     /**
      *
-     * @param requestAuditEvents - Audit search params. Audit Id will be provided by requestAuditEvents.
+     *
+     * @param request - Audit search params. Audit Id will be provided by requestAuditEvents.
      * @return QueryAuditEventsBlobResponse - Response will be having only Audit Blob message.
      *
      * The Adapter logic will be implemented in FHAC-690
      */
-    public QueryAuditEventsBlobResponse queryAuditEventsById(QueryAuditEventsBlobRequest requestAuditEvents) {
-        return new QueryAuditEventsBlobResponse();
+    public QueryAuditEventsBlobResponse queryAuditEventsById(QueryAuditEventsBlobRequest request) {
+        return resultUtil.getQueryAuditEventBlobResponse(getAuditRepositoryDao().queryByAuditId(
+            (new Long(request.getId())).intValue()));
+    }
+
+    /**
+     *
+     * @return AuditRepositoryDAO Instance of AuditRepositoryDAO to query audit events.
+     */
+    protected AuditRepositoryDAO getAuditRepositoryDao() {
```

Comment:
We are using Springg Proxy Configuration files for injection in CONNECT for all other services. To be consistent that's used for AuditQueryLog srevice.